x-pack/plugins/observability/server/lib/annotations/register_annotation_apis.ts

```diff
@@ -56,16 +56,26 @@ export function registerAnnotationAPIs({
         index,
         apiCaller,
         logger,
+        license: context.licensing.license,
       });
 
-      const res = await handler({
-        data: validation.right,
-        client,
-      });
+      try {
+        const res = await handler({
+          data: validation.right,
+          client,
+        });
 
-      return response.ok({
-        body: res,
-      });
+        return response.ok({
+          body: res,
+        });
+      } catch (err) {
+        return response.custom({
+          statusCode: err.output?.statusCode ?? 500,
```

Comment:
Tangential question: is it always better to use nullish coalescing operator (`??`) instead of `||`?
In this case, if `err.output?.statusCode` is `0`, should we return that even though it's an invalid status code, or return 500?

I'm fine with either approach but it would be good to discuss and have some guiding principles for when to use which approach.

Reply:
It's definitely not always better, I agree. In this case my intent is to handle the case that the conditional access operator creates, namely that the value is undefined. If `statusCode` is 0 or `message` is an empty string I feel like that's a bug upstream that we shouldn't cover up. But I can see how that could create confusion for the user.